packages/nitro-protocol/contracts/AssetHolder.sol

```diff
@@ -17,6 +17,128 @@ contract AssetHolder is IAssetHolder {
 
     mapping(bytes32 => bytes32) public assetOutcomeHashes;
 
+
+    // **************
+    // Public methods
+    // **************
+
+    /**
+     * @notice Transfers as many funds escrowed against `channelId` as can be afforded for a specific destination. Assumes no repeated entries.
+     * @dev Transfers as many funds escrowed against `channelId` as can be afforded for a specific destination. Assumes no repeated entries.
+     * @param fromChannelId Unique identifier for state channel to transfer funds *from*.
+     * @param allocationBytes The abi.encode of AssetOutcome.Allocation
+     * @param destination External destination or channel to transfer funds *to*.
+     */
+    function transfer(bytes32 fromChannelId, bytes memory allocationBytes, bytes32 destination) public {
+        // checks
+        _requireCorrectAllocationHash(fromChannelId, allocationBytes);
+        // effects and interactions
+        _transfer(fromChannelId, allocationBytes, destination);
+    }
+
+    /**
+     * @notice Transfers the funds escrowed against `channelId` to the beneficiaries of that channel. Checks against the storage in this contract.
+     * @dev Transfers the funds escrowed against `channelId` and transfers them to the beneficiaries of that channel. Checks against the storage in this contract.
+     * @param channelId Unique identifier for a state channel.
+     * @param allocationBytes The abi.encode of AssetOutcome.Allocation
+     */
+    function transferAll(bytes32 channelId, bytes memory allocationBytes) public override {
+        // checks
+        _requireCorrectAllocationHash(channelId, allocationBytes);
+        // effects and interactions
+        _transferAll(channelId, allocationBytes);
+    }
+
+    /**
+     * @notice Transfers as many funds escrowed against `guarantorChannelId` as can be afforded for a specific destination in the beneficiaries of the __target__ of that channel. Checks against the storage in this contract.
+     * @dev Transfers as many funds escrowed against `guarantorChannelId` as can be afforded for a specific destination in the beneficiaries of the __target__ of that channel. Checks against the storage in this contract.
+     * @param guarantorChannelId Unique identifier for a guarantor state channel.
+     * @param guaranteeBytes The abi.encode of Outcome.Guarantee
+     * @param allocationBytes The abi.encode of AssetOutcome.Allocation for the __target__
+     * @param destination External destination or channel to transfer funds *to*.
+     */
+    function claim(
+        bytes32 guarantorChannelId,
+        bytes memory guaranteeBytes,
+        bytes memory allocationBytes,
+        bytes32 destination
```

Comment:
Idea from @andrewgordstewart: Instead of passing in the `destination`, better to pass in the `index` of the destination in the allocation. Same goes (actually more urgently so) for `transfer`. 

We could even pass `indices: number[]` to target multiple destinations -- and have `[]` imply "all".

Because we pass in the abi-encoded allocation, this implies we know which destination is at which index (so no need to recheck this on chain).